src/xtgeo/xyz/_xyz_roxapi.py

```diff
@@ -281,44 +244,73 @@ def _roxapi_xyz_to_dataframe(
 
         dfr = pd.concat(dfs)
 
-    elif not is_polygons and isinstance(roxxyz, np.ndarray):
+    elif xyztype == _XYZType.POINTS.value and isinstance(roxxyz, np.ndarray):
         # points
         dfr = pd.DataFrame.from_records(roxxyz, columns=XYZ_COLUMNS)
 
     else:
-        raise RuntimeError(f"Unknown error in getting data from Roxar: {type(roxxyz)}")
+        raise RuntimeError(f"Unknown error in getting data from RMS: {type(roxxyz)}")
 
     dfr.reset_index(drop=True, inplace=True)
     return dfr
 
 
+def _infer_type(values: list) -> str:
```

Comment:
Could we do this instead and use it for both polygons and points in the `_add_attributes_to_dataframe` function?
I see from the rmapi documentation that the get_attributes_values returns a numpy masked array, similar as for points.

```python

def _infer_type(values: np.array) -> Literal["str", "float", "int"]:
    """Helper function for points/polygons attributes type."""
    dtype = str(values.dtype)
    if "int" in dtype:
        return "int"
    if "float" in dtype:
        return "float"
    return "str"
 ```

Reply:
Well spotted. I thought values from Polygons were a list, but I see now it is a masked array. So I have adjusted the code.